src/main/java/com/switchfully/parkshark_2024_10/division/Division.java

```diff
@@ -1,14 +1,48 @@
 package com.switchfully.parkshark_2024_10.division;
 
 
-import jakarta.persistence.Entity;
-import jakarta.persistence.Id;
+import com.switchfully.parkshark_2024_10.user.Director;
+import jakarta.persistence.*;
 
 @Entity
 public class Division {
     @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE , generator = "division_seq")
+    @SequenceGenerator(name = "division_seq", sequenceName = "division_seq", allocationSize = 1)
     private Long id;
 
+    @Column(name = "name")
+    private String name;
+
+    @OneToOne
+    @JoinColumn(name = "id")
+    private Director director;
```

Comment:
I think this needs to be more verbose (add targetEntity for example). Also, it should probably use the `mapped_by` referencing the `Director.division` relation. Or vice versa if you prefer.

Reply:
not sure why this needs to be more verbose?